### PR TITLE
*: replace []*Refs with []Refs

### PR DESCRIPTION
**Pull request by gyuho**
Address https://github.com/kubernetes/test-infra/issues/9950.

/cc @fejta

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
Why?

/cc @kargakis

**Comment by gyuho** [comment]
@stevekuznetsov See https://github.com/kubernetes/test-infra/issues/9950.

File: prow/apis/prowjobs/v1/types.go [hunk 1]
```diff
@@ -108,10 +108,10 @@ type ProwJobSpec struct {
 	Job string `json:"job,omitempty"`
 	// Refs is the code under test, determined at
 	// runtime by Prow itself
-	Refs *Refs `json:"refs,omitempty"`
+	Refs Refs `json:"refs,omitempty"`
```
Comment (fejta) [inline prow/apis/prowjobs/v1/types.go]: Unfortunately this actually should remain as `*Refs`. We want to distinguish unset from unconfigured (and we do not want every prow job to always output the Refs field.

**Comment by gyuho** [comment]
@stevekuznetsov @fejta PTAL. Thanks!

**Comment by fejta** [comment]
/lgtm

Thanks so much!

File: prow/clonerefs/run.go [hunk 2]
```diff
@@ -73,7 +73,7 @@ func (o Options) Run() error {
 	}
 
 	for _, ref := range o.GitRefs {
-		input <- ref
+		input <- &ref
```
Comment (cjwagner) [inline prow/clonerefs/run.go]: This is incorrect, this sends the same pointer to input multiple times. ref #10093

Comment (fejta) [reply prow/clonerefs/run.go]: Agree with Ben, nice catch... can we add a unit test that would have detected this problem?

**Pull request merged**